[Bug]: [Nightly] Some cases related to compaction failed

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:5141e05
- Deployment mode(standalone or cluster):cluster
- SDK version(e.g. pymilvus v2.0.0rc2):2.2.0.dev30
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```


### Current Behavior

[2022-09-20T21:08:05.850Z] [gw2] [ 11%] FAILED testcases/test_compaction.py::TestCompactionOperation::test_compact_less_threshold_no_merge
[2022-09-20T21:08:30.494Z] [gw1] [ 12%] FAILED testcases/test_compaction.py::TestCompactionOperation::test_compact_no_merge
[2022-09-20T21:10:38.016Z] [gw4] [ 15%] FAILED testcases/test_compaction.py::TestCompactionParams::test_compact_max_time_interval

### Expected Behavior

_No response_

### Steps To Reproduce

_No response_

### Milvus Log

_No response_

### Anything else?

_No response_

--- a/tests/python_client/testcases/test_compaction.py
+++ b/tests/python_client/testcases/test_compaction.py
@@ -322,6 +322,9 @@ class TestCompactionParams(TestcaseBase):
 
         sleep(ct.max_compaction_interval + 1)
 
+        # create index
+        collection_w.create_index(ct.default_float_vec_field_name, ct.default_index)
+
         # verify queryNode load the compacted segments
         collection_w.load()
         replicas = collection_w.get_replicas()[0]
@@ -831,6 +834,7 @@ class TestCompactionOperation(TestcaseBase):
         collection_w.insert(df)
         assert collection_w.num_entities == tmp_nb
 
+        collection_w.create_index(ct.default_float_vec_field_name, ct.default_index)
         collection_w.load()
 
         seg_before, _ = self.utility_wrap.get_query_segment_info(collection_w.name)
@@ -1011,6 +1015,9 @@ class TestCompactionOperation(TestcaseBase):
 
         # create collection shard_num=1, insert 9 segments, each with one entity
         collection_w = self.collection_insert_multi_segments_one_shard(prefix, num_of_segment=less_threshold)
+
+        # create index
+        collection_w.create_index(ct.default_float_vec_field_name, ct.default_index)
 
         # load and verify no auto-merge
         collection_w.load()

